scooby/report.py

```diff
@@ -454,56 +436,40 @@ def get_version(module):
     # If (1), we have to load it, if (2), we have to get its name.
     if isinstance(module, str):  # Case 1: module is a string; import
         name = module  # The name is stored in module in this case.
+    elif isinstance(module, ModuleType):  # Case 2: module is module; get name
+        name = module.__name__
+    else:  # If not str nor module raise error
+        raise TypeError("Cannot fetch version from type " "({})".format(type(module)))
 
-        # Import module `name`; set to None if it fails.
+    # Use importlib.metadata to get the version
+    try:
+        ver = importlib.metadata.version(name)
+    except (importlib.metadata.PackageNotFoundError):  # pragma: no cover
+        ver = None  # Package be be in PATH
+    except:  # noqa
+        return name, MODULE_TROUBLE
+
+    if ver is None:
+        # Handle scenario when package is on path but not installed
+        # `importlib.metadata.version` should handle this for py3.11+
         try:
             module = importlib.import_module(name)
         except ImportError:
-            module = None
+            return name, MODULE_NOT_FOUND
         except:  # noqa
             return name, MODULE_TROUBLE
-
-    elif isinstance(module, ModuleType):  # Case 2: module is module; get name
-        name = module.__name__
-
-    else:  # If not str nor module raise error
-        raise TypeError("Cannot fetch version from type " "({})".format(type(module)))
-
-    # Now get the version info from the module
-    if module is None:
-        ver = pkg_resources_version_fallback(name)
-        if ver is not None:
-            return name, ver
-        return name, MODULE_NOT_FOUND
-    else:
-        # Try common version names.
+        # Try common version names
         for v_string in ('__version__', 'version'):
             try:
-                return name, getattr(module, v_string)
+                ver = getattr(module, v_string)
+                break
             except AttributeError:
                 pass
-
-        # Try the VERSION_ATTRIBUTES library
-        try:
-            attr = VERSION_ATTRIBUTES[name]
-            return name, getattr(module, attr)
-        except (KeyError, AttributeError):
-            pass
-
-        # Try the VERSION_METHODS library
-        try:
-            method = VERSION_METHODS[name]
-            return name, method()
-        except (KeyError, ImportError):
-            pass
-
-        # Try package-resource distribution version
-        ver = pkg_resources_version_fallback(name)
         if ver is not None:
-            return name, ver
+            return name, f'{ver} {NOT_PROPERLY_INSTALLED}'
+        return name, f'{VERSION_NOT_FOUND} {NOT_PROPERLY_INSTALLED}'
```

Comment:
I am always reluctant too have to many returns, but that is personal preference. What about
```
        if ver is None:
            ver = VERSION_NOT_FOUND
        ver += f' {NOT_PROPERLY_INSTALLED}'
```
and then it is returned at the end?